src/highlighters.cc

```diff
@@ -1130,52 +1130,56 @@ constexpr StringView LineNumbersHighlighter::ms_id;
 void show_matching_char(HighlightContext context, DisplayBuffer& display_buffer, BufferRange)
 {
     const Face face = context.context.faces()["MatchingChar"];
-    using CodepointPair = std::pair<Codepoint, Codepoint>;
-    static const CodepointPair matching_chars[] = { { '(', ')' }, { '{', '}' }, { '[', ']' }, { '<', '>' } };
+    // static const CodepointPair matching_chars[] = { { '(', ')' }, { '{', '}' }, { '[', ']' }, { '<', '>' } };
+    const auto& matching_pairs = context.context.options()["matching_pairs"].get<Vector<Codepoint, MemoryDomain::Options>>();
     const auto range = display_buffer.range();
     const auto& buffer = context.context.buffer();
     for (auto& sel : context.context.selections())
     {
         auto pos = sel.cursor();
         if (pos < range.begin or pos >= range.end)
             continue;
-        auto c = buffer.byte_at(pos);
-        for (auto& pair : matching_chars)
+        auto match = matching_pairs.end();
+        match = find(matching_pairs, buffer.byte_at(sel.cursor()));
```

Comment:
Look for the codepoint, not the byte, so that arbitrary matching pairs can be used

Reply:
Am I right that dereferencing a buffer iterator at the cursor position would do that? I admit that I have not looked too closely at the internal buffer manipulation.